examples/self-hosted-shared-cache/README.md

```diff
@@ -0,0 +1,45 @@
+# Example of the self-hosted app with Redis as cache storage
+
+This example demonstrates using Redis as a shared cache for hosting multiple instances of a Next.js app. It supports caching at both the App and Pages routes in default and standalone modes, as well as Partial Pre-rendering. This functionality is made possible by the [`@neshca/cache-handler`](https://github.com/caching-tools/next-shared-cache/tree/canary/packages/cache-handler) package, which offers an API to customize cache handlers and seamlessly replace Next.js' default cache.
```

Comment:
Could you explain why the custom package is needed versus just putting it into this example directly?

Reply:
TLDR: this custom package takes on the responsibilities of the original class, providing a concise API for the user.

Since the API implies replacing the original FileSystemCache class with a custom one, it is not enough to replace the Map with the Redis Client from [the example in the docs](https://nextjs.org/docs/app/api-reference/next-config-js/incrementalCacheHandlerPath). In this case, reading pre-rendered pages from disk will be lost. You can see [that my test is failing](https://github.com/caching-tools/next-shared-cache/actions/runs/6764512665/job/18382888425?pr=94#step:9:240) if I restrict file system access. The library implements work with the file system. So, when Redis fails, the applications in pods will continue to work with the file system until Redis is online.

Also, for the correct operation of on-demand revalidation, you must have a common TagsManifest for all pods. Since tagsManifest is checked in [this](https://github.com/vercel/next.js/blob/31c2f976cdd4deb0e7c412538a70b795dff4689e/packages/next/src/server/lib/incremental-cache/file-system-cache.ts#L255) and [this](https://github.com/vercel/next.js/blob/31c2f976cdd4deb0e7c412538a70b795dff4689e/packages/next/src/server/lib/incremental-cache/file-system-cache.ts#L275) places in the original class, it is necessary to support this functionality.

This functionality is the same for everyone and is, therefore, encapsulated in the library.

In an ideal world, I would like to have such an API for customizing caching out of the box, and I would be willing to spend the effort and time to implement this for Next.js since, in essence, my library is inspired mainly by your implementation.

Reply:
I will add this to example

Reply:
I also created this issue #58094. For now, it is crucial to read from disk using cache. This custom package deals with this out of the box.

If the disk is being used, key eviction alone cannot make on-demand revalidation; the users must also delete files. I hope this limitation is addressed soon, but now it works like so.